### PR TITLE
*: support put with lease

### PR DESCRIPTION
**Pull request by xiang90**
Now we persist the lease attached to each kv.

### CODE REVIEW

**Comment by gyuho** [comment]
I like that we now have type for ID. We should have it for watchID as well. Think @heyitsanthony is working on it?

File: storage/kvstore.go [hunk 1]
```diff
@@ -39,6 +39,8 @@ var (
 	markBytePosition       = markedRevBytesLen - 1
 	markTombstone     byte = 't'
 
+	noLease = LeaseID(0)
```
Comment (heyitsanthony) [inline storage/kvstore.go]: Probably should be `NoLease` since it has meaning w/r/t the public KV interface

Comment (xiang90) [inline storage/kvstore.go]: Sure.

**Comment by heyitsanthony** [comment]
Seems straightforward. lgtm once CI passes and `NoLease` happens

**Comment by gyuho** [comment]
LGTM, after fixing benchmark errors.

**Pull request merged**